From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Lemures Lemniscati <lemures.lemniscati@gmail.com>
Date: Sat, 17 Oct 2020 13:40:41 +0900
Subject: [PATCH] 1.6 Makefile cygwin


diff --git a/Makefile b/Makefile
index 43f8273..1c1cdd5 100644
--- a/Makefile
+++ b/Makefile
@@ -104,6 +104,12 @@ else ifeq ($(shell uname),Darwin)
    SHARED_LIB_CFLAGS  := -fPIC
    SHARED_LIB_LDFLAGS := -install_name $(SHARED_LIB)
 
+# Cygwin
+else ifeq ($(shell uname -o),Cygwin)
+   SHARED_LIB         := cygdeflate-$(SOVERSION).dll
+   SHARED_LIB_LDFLAGS := -Wl,--out-implib=libdeflate.dll.a \
+                         -Wl,--export-all-symbols
+   PROG_SUFFIX        := .exe
 # Linux, FreeBSD, etc.
 else
    SHARED_LIB         := libdeflate.so.$(SOVERSION)
@@ -251,11 +257,11 @@ $(ALL_PROG_OBJ): %.o: %.c $(ALL_PROG_COMMON_HEADERS) $(COMMON_HEADERS) \
 # test programs must be linked with zlib for doing comparisons.
 
 $(NONTEST_PROGRAMS): %$(PROG_SUFFIX): programs/%.o $(PROG_COMMON_OBJ) \
-			$(STATIC_LIB)
+                       $(SHARED_LIB)
 	$(QUIET_CCLD) $(CC) -o $@ $(LDFLAGS) $(PROG_CFLAGS) $+
 
 $(TEST_PROGRAMS): %$(PROG_SUFFIX): programs/%.o $(PROG_COMMON_OBJ) \
-			$(TEST_PROG_COMMON_OBJ) $(STATIC_LIB)
+			$(TEST_PROG_COMMON_OBJ) $(SHARED_LIB)
 	$(QUIET_CCLD) $(CC) -o $@ $(LDFLAGS) $(PROG_CFLAGS) $+ -lz
 
 ifdef HARD_LINKS
@@ -286,9 +292,10 @@ all:$(DEFAULT_TARGETS)
 # '-D' and '-t' options, so don't use them; use portable commands only.
 install:all
 	install -d $(DESTDIR)$(LIBDIR) $(DESTDIR)$(INCDIR) $(DESTDIR)$(BINDIR)
-	install -m644 $(STATIC_LIB) $(DESTDIR)$(LIBDIR)
-	install -m755 $(SHARED_LIB) $(DESTDIR)$(LIBDIR)
-	ln -sf $(SHARED_LIB) $(DESTDIR)$(LIBDIR)/libdeflate.so
+	#install -m644 $(STATIC_LIB) $(DESTDIR)$(LIBDIR)
+	install -m755 $(SHARED_LIB) $(DESTDIR)$(BINDIR)
+	#ln -sf $(SHARED_LIB) $(DESTDIR)$(LIBDIR)/libdeflate.so
+	install -m644 libdeflate.dll.a $(DESTDIR)$(LIBDIR)/
 	install -m644 libdeflate.h $(DESTDIR)$(INCDIR)
 	install -m755 gzip $(DESTDIR)$(BINDIR)/libdeflate-gzip
 	ln -f $(DESTDIR)$(BINDIR)/libdeflate-gzip $(DESTDIR)$(BINDIR)/libdeflate-gunzip
